src/lib/shim/shim_sys.c

```diff
@@ -30,6 +30,20 @@ static EmulatedTime _shim_sys_get_time() {
 
 uint64_t shim_sys_get_simtime_nanos() { return _shim_sys_get_time() / SIMTIME_ONE_NANOSECOND; }
 
+static SimulationTime _shim_sys_latency_for_syscall(long n) {
+    switch (n) {
+        case SYS_clock_gettime:
+        case SYS_time:
+        case SYS_gettimeofday:
```

Comment:
This doesn't make the distinction between syscalls and the libc VDSO wrappers. For example, my understanding is that if you call `syscall(SYS_clock_gettime)`, it won't be handled by the VDSO and should incur the full syscall latency penalty. I don't think Shadow needs to handle that correctly, but I think there should be a comment here explaining how shadow's behaviour differs from linux.

Also, shadow still has a `syscallhandler_clock_gettime()` implementation. Is there still any path where this function gets called? If not should we remove it? And if there is, should we also apply the VDSO latency from the shadow side?

Reply:
> I think there should be a comment here explaining how shadow's behaviour differs from linux.

Done

> Also, shadow still has a syscallhandler_clock_gettime() implementation. Is there still any path where this function gets called? 

It'll still get called in ptrace mode, if the managed code didn't call the `clock_gettime` libc function. We can probably safely delete it when we remove ptrace mode.

> And if there is, should we also apply the VDSO latency from the shadow side?

In the cases where it'd happen, charging the syscall latency is actually more-correct :)